src/components/InfiniteLoader.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import * as React from 'react';
+import { action, observable } from 'mobx';
+import { observer } from 'mobx-react';
+import { ActivityIndicator, Platform } from 'react-native';
+
+type Props = {|
+  fetchNextData: () => Promise<void>,
+|};
+
+export type InfiniteLoaderChildProps = {
+  loadingIndicator: React.Node,
+  onEndReached: () => Promise<void>,
+  onEndReachedThreshold: number,
+};
+
+@observer
+class InfiniteLoader extends React.Component<Props> {
+  @observable _loading = false;
+
+  @action
+  _setLoading = (loading: boolean) => {
+    this._loading = loading;
+  };
+
+  componentWillMount() {
+    this._fetchNextData();
+  }
+
+  _loadingIndicator = (): React.Node =>
```

Comment:
Just inline this. No sub-render functions

Reply:
so, you prefer that I propogate `loading` bool, instead this render fucntion to children? I made it to avoiding boilerplating `ListFooterComponent` of FlatList on every component